akka-projection-kafka/src/it/scala/akka/projection/kafka/KafkaSourceProviderSpec.scala

```diff
@@ -8,22 +8,22 @@ import java.lang.{ Long => JLong }
 
 import scala.concurrent.Await
 import scala.concurrent.Future
+import scala.concurrent.duration._
 
 import akka.actor.typed.scaladsl.adapter._
 import akka.projection.MergeableOffset
+//import akka.projection.kafka.internal.KafkaSourceProviderImpl
 import akka.projection.kafka.scaladsl.KafkaSourceProvider
 import akka.stream.scaladsl.Source
 import akka.stream.testkit.scaladsl.StreamTestKit.assertAllStagesStopped
 import akka.stream.testkit.scaladsl.TestSink
-import org.scalatest.Ignore
 
-@Ignore
-class KafkaSourceProviderSpec extends KafkaSpecBase() {
+class KafkaSourceProviderSpec extends KafkaSpecBase {
   "KafkaSourceProviderSpec" must {
     "resume from provided offsets" in assertAllStagesStopped {
       val topic = createTopic()
       val groupId = createGroupId()
-      val settings = consumerDefaults().withGroupId(groupId)
+      val settings = consumerDefaults().withGroupId(groupId).withStopTimeout(0.seconds)
```

Comment:
I assuem `withStopTimeout(0.seconds)` disables the timeout (?). I guess it is to make sure this stop timeout doesn't kick in before other timeouts (retries, etc...) in the test.
It'd be great to add a comment indicating why.

Reply:
The reason I added this is here: https://github.com/akka/akka-projection/pull/340/files#r454353872 .  I'll add a comment.  It's purpose is to delay shutting down the source to let in-flight messages complete.  We don't currently have a concept of shutting down a source provider gracefully, but if we did then this config would actually be in the impl, along with a call to the Alpakka Kafka `DrainingControl`.

I'll create a ticket for source provider graceful shutdown.

Reply:
https://github.com/akka/akka-projection/issues/342